sim/core/character.go

```diff
@@ -727,3 +757,45 @@ func FillTalentsProto(data protoreflect.Message, talentsStr string, treeSizes [3
 		offset += treeSizes[treeIdx]
 	}
 }
+
+func (character *Character) applyArmorSpecialization() bool {
+	hasBonus := true
+
+	var armorSpecType proto.ArmorType
+
+	if character.ArmorSpecialization == proto.ArmorSpecialization_ArmorSpecializationCloth {
+		armorSpecType = proto.ArmorType_ArmorTypeCloth
+	} else if character.ArmorSpecialization == proto.ArmorSpecialization_ArmorSpecializationLeather {
+		armorSpecType = proto.ArmorType_ArmorTypeLeather
+	} else if character.ArmorSpecialization == proto.ArmorSpecialization_ArmorSpecializationMail {
+		armorSpecType = proto.ArmorType_ArmorTypeMail
+	} else if character.ArmorSpecialization == proto.ArmorSpecialization_ArmorSpecializationPlate {
+		armorSpecType = proto.ArmorType_ArmorTypePlate
+	}
```

Comment:
Instead of having a separate `proto.ArmorSpecialization` enum just use `proto.ArmorType`